lib/driver.dart

```diff
@@ -0,0 +1,407 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
```

Comment:
There's a specific protocol between `driver.dart` and the local server. I'm afraid that it may cost more later to change this protocol because changing it requires changes in two repositories and there's no way to make the change in a single PR atomically. I see two options:

1. This package does not include a local server and only provides the `Recorder` API. The client/server protocol will continue living over in the flutter/flutter repo.
2. This package includes both the client and server implementation. The code in flutter/flutter would import both. The server API would not depend on/implement the "task" protocol used by the devicelab. Instead it would provide an API for collecting all the benchmark numbers. The devicelab task will be responsible for converting that to the devicelab protocol.

Reply:
It just occurred to me that option 2 has the disadvantage that this package would need more dependencies in order to implement the server (`package:shelf` and friends). Your current approach and option 2 doesn't have that problem.

Reply:
I think it's good to keep it as is. It is definitely possible to reduce duplicate code by creating a common package for the server, but that seems unnecessary at this time.